funsor/ops.py

```diff
@@ -28,6 +28,34 @@ def __str__(self):
         return self.__name__
 
 
+class TransformOp(Op):
+    def set_inv(self, fn):
+        """
+        :param callable fn: A function that inputs an arg ``y`` and outputs a
+            value ``x`` such that ``y=self(x)``.
+        """
+        assert callable(fn)
+        self.inv = fn
+        return fn
+
+    def set_log_abs_det_jacobian(self, fn):
+        """
+        :param callable fn: A function that inputs two args ``x, y``, where
+            ``y=self(x)``, and returns ``log(abs(det(dy/dx)))``.
+        """
+        assert callable(fn)
+        self.log_abs_det_jacobian = fn
+        return fn
+
+    @staticmethod
+    def inv(x):
+        raise NotImplementedError
+
+    @staticmethod
+    def log_abs_det_jacobian(x, y):
+        raise NotImplementedError
```

Comment:
Do `inv` and `log_abs_det_jacobian` need to be methods, as opposed to living in tables like other facts about `Op`s?  Is this so we can support more advanced monolithic transforms later on?

Reply:
I guess the could live in tables. The method interface is closer to `torch.distributions.Transform`, which makes it a little more familiar.

Reply:
I imagine more complex transform later using `@solve.register` rather than this `TransformOp` interface. I'm not set on any of the interfaces in this PR, and I assume we'll readjust once `Joint` is more flexible and we can write more examples and tests.

Reply:
Ok, I'll go ahead and merge and we can revisit questions about interfaces when `Joint` is ready for it.